### PR TITLE
Enable option for patch method to send a diff

### PR DESCRIPTION
**Pull request by wdmtech**
Relating to #148 

Here's my first attempt at getting this working.

Cheers!

@marshallswain

### CODE REVIEW

File: test/service-module/actions.test.js [hunk 1]
```diff
@@ -537,6 +537,32 @@ describe('Service Module - Actions', () => {
...
 
   describe('Patch', () => {
+    it('updates only the changed properties', (done) => {
+      const store = new Vuex.Store({
+        plugins: [service('todos')]
+      })
+      const todoState = store.state.todos
+      const actions = mapActions('todos', ['create', 'patch'])
+
+      actions.create.call({$store: store}, {description: 'Do the second', unchanged: true, deep: {changed: false, unchanged: true}})
```
Comment (marshallswain) [inline test/service-module/actions.test.js]: To make it cleaner and more clear, please put the data here into a variable and pass it in.

**Comment by marshallswain** [comment]
Lovely work.  I just had one comment.  Please clean that up and I will get this merged in!  I'm going to make this a new major release.

File: test/service-module/actions.test.js (hunk 1 above)
Comment (wdmtech) [reply test/service-module/actions.test.js]: Something like this?

```js
// ...
const dataUnchanged = {unchanged: true, deep: {changed: false, unchanged: true}}
const dataChanged = {unchanged: true, deep: {changed: true, unchanged: true}}

actions.create.call({$store: store}, Object.assign({description: 'Do the second'}, dataUnchanged)
  .then(response => {
    actions.patch.call({$store: store}, [0, Object.assign({description: 'Write a Vue app'}, dataChanged)])
      .then(responseFromPatch => {
        assert(todoState.ids.length === 1)
        assert(todoState.errorOnPatch === null)
        assert(todoState.isPatchPending === false)
        assert.deepEqual(todoState.keyedById[responseFromPatch.id], responseFromPatch)
        done()
      })
```

**Comment by marshallswain** [comment]
@wdmtech I'm going to merge this and turn it off by default so it's not a breaking change.  Thank you for the awesome PR.

**Comment by wdmtech** [comment]
My pleasure! Glad I could help with this great library 😀

**Pull request merged**